### ISSUE TITLE
`ideviceinstaller list` with `--xml` returns 1 after all info printed

### ISSUE THREAD

**Issue by tomriddly** [libimobiledevice/ideviceinstaller#155]
I'm using ideviceinstaller compiled with [835e626f1413a520093a5f1fa5c797016c13d80f](https://github.com/libimobiledevice/ideviceinstaller/tree/835e626f1413a520093a5f1fa5c797016c13d80f) on macos 13.4.1 (c) (22F770820d). `list` commands works fine and output infos as expected. 
But if I use `--xml` as parameter, all info are printed as well but exit code of `ideviceinstaller` becomes "1" which is so wired. No other error info are printed.
```shell
# ideviceinstaller --version # version of ideviceinstaller
ideviceinstaller 1.1.1-23-g835e626
# idevice_id -v # version of libimobiledevice
idevice_id 1.3.0-190-g6fc41f5
# plistutil -v # version of libplist
plistutil 2.3.0-18-g2d8d7ef
# iproxy -v # version of libusbmuxd
iproxy 2.0.2-31-g8d30a55
# uname -a  
Darwin B-P4LFMD6R-2158.local 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64 x86_64
```

You can reproduce this by following command. With or without `-b` will not change the exit code. 
```shell
ideviceinstaller -u 00008101-001238120228001E list --all --xml -b com.apple.MobileStore
echo $?
```

**Comment by tomriddly** [comment]
it seems that if output format is specific, res is not assigned before `goto leave_cleanup;`
https://github.com/libimobiledevice/ideviceinstaller/blob/835e626f1413a520093a5f1fa5c797016c13d80f/src/ideviceinstaller.c#L928

**Comment by nikias** [comment]
Fixed with ab26b03baf862157900c9b9be35b382ddc7e245d.
